### DISCUSSION TITLE
alarm 8 while homing GrblHAL 1.1f

### DISCUSSION THREAD

**Discussion by jrdarrah** [terjeio/grblHAL#291] [Q&A]
I'm setting us a new CNC mill using  the grblHAL breakout board. I've been taking small steps and made sure the steppers are working properly.  I added NPN Normally Open proximity switches, two per axis and verified that all are at about 5V when not triggered and .7V when triggered.  For this test I have all three axis in about the middle so none of the sensors are triggered.  To keep from driving the motors to the end stops if the sensors don't work I disconnected the steppers  with my plan to manually trigger the end sensor.  I can see alarms when I trigger the sensor manually from reset/unlock.

The problem now is when I try homing it immediately gets this error. the UGS display doesn't indicate  any movement on any of the axis. The error doesn't make sense because the sensors are not triggered  so they wouldn't pull off.  In my test I will manually trigger which I didn't. I was expecting UGS/grblHAL to count up/down on all three axises and back off when I manually triggered.

I did look at the $5 invert limit pins but I think it is correct at $5=0 since $5V is not triggered and .7V is.

Have I missed something setting up the homing in grblHAL? 

>>> $H
[Error] An error was detected while sending '$H': (ALARM:8) Homing fail. Pull off travel failed to clear limit switch. Try increasing pull-off setting or check wiring. Streaming has been paused.

[Error] Error while processing response <An error was detected while sending '$H': (ALARM:8) Homing fail. Pull off travel failed to clear limit switch. Try increasing pull-off setting or check wiring. Streaming has been paused.>

[Error] Error while processing response <ok>: An unexpected command was completed by the controller.

GrblHAL 1.1f ['$' for help]
>>> $$
$0 = 10.0    (Step pulse time, microseconds)
$1 = 25    (Step idle delay, milliseconds)
$2 = 7    (Step pulse invert, mask)
$3 = 0    (Step direction invert, mask)
$4 = 7    (Invert step enable pin, boolean)
$5 = 0    (Invert limit pins, boolean)
$6 = 0    (Invert probe pin, boolean)
$10 = 18    (Status report options, mask)
$11 = 0.010    (Junction deviation, millimeters)
$12 = 0.002    (Arc tolerance, millimeters)
$13 = 0    (Report in inches, boolean)
$14 = 64   
$15 = 0   
$16 = 0   
$17 = 0   
$18 = 0   
$19 = 0   
$20 = 0    (Soft limits enable, boolean)
$21 = 1    (Hard limits enable, boolean)
$22 = 1    (Homing cycle enable, boolean)
$23 = 0    (Homing direction invert, mask)
$24 = 25.000    (Homing locate feed rate, mm/min)
$25 = 500.000    (Homing search seek rate, mm/min)
$26 = 25    (Homing switch debounce delay, milliseconds)
$27 = 1.000    (Homing switch pull-off distance, millimeters)
$28 = 0.100   
$29 = 0.0   
$30 = 1000.    (Maximum spindle speed, RPM)
$31 = 0.    (Minimum spindle speed, RPM)
$32 = 0    (Laser-mode enable, boolean)
$33 = 5000.000   
$34 = 0.000   
$35 = 0.000   
$36 = 100.000   
$37 = 0   
$39 = 1   
$40 = 0   
$41 = 0   
$42 = 2   
$43 = 1   
$44 = 4   
$45 = 3   
$46 = 0   
$56 = 5.000   
$57 = 100.000   
$58 = -5.000   
$59 = 500.000   
$60 = 0   
$61 = 0   
$62 = 0   
$63 = 2   
$64 = 0   
$65 = 0   
$70 = 3   
$100 = 158.302    (X-axis travel resolution, step/mm)
$101 = 159.400    (Y-axis travel resolution, step/mm)
$102 = 159.453    (Z-axis travel resolution, step/mm)
$110 = 500.000    (X-axis maximum rate, mm/min)
$111 = 500.000    (Y-axis maximum rate, mm/min)
$112 = 800.000    (Z-axis maximum rate, mm/min)
$120 = 10.000    (X-axis acceleration, mm/sec^2)
$121 = 10.000    (Y-axis acceleration, mm/sec^2)
$122 = 10.000    (Z-axis acceleration, mm/sec^2)
$130 = 200.000    (X-axis maximum travel, millimeters)
$131 = 200.000    (Y-axis maximum travel, millimeters)
$132 = 200.000    (Z-axis maximum travel, millimeters)
$300 = GRBL   
$301 = 0   
$302 = 10.1.1.2   
$303 = 10.1.1.1   
$304 = 255.255.255.0   
$305 = 23   
$307 = 80   
$341 = 0   
$342 = 30.0   
$343 = 25.0   
$344 = 200.0   
ok
>>> $G
[GC:G0 G54 G17 G21 G90 G94 G49 G98 G50 M5 M9 T0 F0 S0.]
ok

**Comment by terjeio** [comment]
What is reported in the `Pn:` element of the realtime report before and after a limit switch is triggered?
Example when not triggered:
`<Idle|MPos:0.000,0.000,0.000|Bf:511,1023|FS:0,0,118>`
When X is triggered:
`<Idle|MPos:0.000,0.000,0.000|Bf:511,1023|FS:0,0,118|Pn:X>`

**Reply by jrdarrah** [reply]
I ran 4 tests, reset-unlock trigger X, Y, Z axis sensor one at a time, then a reset unlock and home
I did some editing to reduce repeated data so hopefully this caught what you asked for

**** Connected to /dev/ttyACM0 @ 115200 baud ****

GrblHAL 1.1f ['$' for help]
>>> $$
$0 = 10.0    (Step pulse time, microseconds)
$1 = 25    (Step idle delay, milliseconds)
$2 = 7    (Step pulse invert, mask)
$3 = 0    (Step direction invert, mask)
$4 = 7    (Invert step enable pin, boolean)
$5 = 0    (Invert limit pins, boolean)
$6 = 0    (Invert probe pin, boolean)
$10 = 2    (Status report options, mask)
$11 = 0.010    (Junction deviation, millimeters)
$12 = 0.002    (Arc tolerance, millimeters)
$13 = 0    (Report in inches, boolean)
$14 = 64   
$15 = 0   
$16 = 0   
$17 = 0   
$18 = 0   
$19 = 0   
$20 = 0    (Soft limits enable, boolean)
$21 = 1    (Hard limits enable, boolean)
$22 = 1    (Homing cycle enable, boolean)
$23 = 0    (Homing direction invert, mask)
$24 = 25.000    (Homing locate feed rate, mm/min)
$25 = 500.000    (Homing search seek rate, mm/min)
$26 = 25    (Homing switch debounce delay, milliseconds)
$27 = 1.000    (Homing switch pull-off distance, millimeters)
$28 = 0.100   
$29 = 0.0   
$30 = 1000.    (Maximum spindle speed, RPM)
$31 = 0.    (Minimum spindle speed, RPM)
$32 = 0    (Laser-mode enable, boolean)
$33 = 5000.000   
$34 = 0.000   
$35 = 0.000   
$36 = 100.000   
$37 = 0   
$39 = 1   
$40 = 0   
$41 = 0   
$42 = 2   
$43 = 1   
$44 = 4   
$45 = 3   
$46 = 0   
$56 = 5.000   
$57 = 100.000   
$58 = -5.000   
$59 = 500.000   
$60 = 0   
$61 = 0   
$62 = 0   
$63 = 2   
$64 = 0   
$65 = 0   
$70 = 3   
$100 = 158.302    (X-axis travel resolution, step/mm)
$101 = 159.400    (Y-axis travel resolution, step/mm)
$102 = 159.453    (Z-axis travel resolution, step/mm)
$110 = 500.000    (X-axis maximum rate, mm/min)
$111 = 500.000    (Y-axis maximum rate, mm/min)
$112 = 800.000    (Z-axis maximum rate, mm/min)
$120 = 10.000    (X-axis acceleration, mm/sec^2)
$121 = 10.000    (Y-axis acceleration, mm/sec^2)
$122 = 10.000    (Z-axis acceleration, mm/sec^2)
$130 = 200.000    (X-axis maximum travel, millimeters)
$131 = 200.000    (Y-axis maximum travel, millimeters)
$132 = 200.000    (Z-axis maximum travel, millimeters)
$300 = GRBL   
$301 = 0   
$302 = 10.1.1.2   
$303 = 10.1.1.1   
$304 = 255.255.255.0   
$305 = 23   
$307 = 80   
$341 = 0   
$342 = 30.0   
$343 = 25.0   
$344 = 200.0   
ok
>>> $G
[verbose] GrblFeedbackMessage{message='[GC:G0 G54 G17 G21 G90 G94 G49 G98 G50 M5 M9 T0 F0 S0.]', distanceMode='G90', units='G21'}
[GC:G0 G54 G17 G21 G90 G94 G49 G98 G50 M5 M9 T0 F0 S0.]
ok
[verbose] <Alarm|WPos:0.000,0.000,0.000|Bf:35,1023|H:0>
[verbose] <Alarm|WPos:0.000,0.000,0.000|Bf:35,1023>



>>> $X
[MSG:Caution: Unlocked]
ok


[verbose] <Idle|WPos:0.000,0.000,0.000|Bf:35,1023>

****Trigger X-axis sensor****
An unexpected error was detected: (ALARM:1) Hard limit has been triggered. Machine position is likely lost due to sudden halt. Re-homing is highly recommended.
[MSG:Reset to continue]

[verbose] <Alarm|WPos:0.000,0.000,0.000|Bf:35,1023>
 
GrblHAL 1.1f ['$' for help]
>>> $$
//edited//

>>> $G
[verbose] GrblFeedbackMessage{message='[GC:G0 G54 G17 G21 G90 G94 G49 G98 G50 M5 M9 T0 F0 S0.]', distanceMode='G90', units='G21'}
[GC:G0 G54 G17 G21 G90 G94 G49 G98 G50 M5 M9 T0 F0 S0.]
ok
[verbose] <Alarm|WPos:0.000,0.000,0.000|Bf:35,1023|H:0>
[verbose] <Alarm|WPos:0.000,0.000,0.000|Bf:35,1023>
 
>>> $X
[MSG:Caution: Unlocked]
ok
[verbose] <Idle|WPos:0.000,0.000,0.000|Bf:35,1023>

Trigger Y-Axis sensor
An unexpected error was detected: (ALARM:1) Hard limit has been triggered. Machine position is likely lost due to sudden halt. Re-homing is highly recommended.
[MSG:Reset to continue]
[verbose] <Alarm|WPos:0.000,0.000,0.000|Bf:35,1023>
 
GrblHAL 1.1f ['$' for help]
>>> $$
//edited//

>>> $G
[verbose] GrblFeedbackMessage{message='[GC:G0 G54 G17 G21 G90 G94 G49 G98 G50 M5 M9 T0 F0 S0.]', distanceMode='G90', units='G21'}
[GC:G0 G54 G17 G21 G90 G94 G49 G98 G50 M5 M9 T0 F0 S0.]
ok
[verbose] <Alarm|WPos:0.000,0.000,0.000|Bf:35,1023|H:0>
[verbose] <Alarm|WPos:0.000,0.000,0.000|Bf:35,1023>
 
>>> $X
[MSG:Caution: Unlocked]
ok
[verbose] <Idle|WPos:0.000,0.000,0.000|Bf:35,1023>


Trigger Z-axis sensor
An unexpected error was detected: (ALARM:1) Hard limit has been triggered. Machine position is likely lost due to sudden halt. Re-homing is highly recommended.
[MSG:Reset to continue]
[verbose] <Alarm|WPos:0.000,0.000,0.000|Bf:35,1023>



GrblHAL 1.1f ['$' for help]
>>> $$
//edited//

>>> $G
[GC:G0 G54 G17 G21 G90 G94 G49 G98 G50 M5 M9 T0 F0 S0.]
ok
>>> $X
[MSG:Caution: Unlocked]
ok
[verbose] <Idle|WPos:0.000,0.000,0.000|Bf:35,1023>

Home sequence
>>> $H
[verbose] <Home|WPos:0.000,0.000,0.000|Bf:35,1023>
[Error] An error was detected while sending '$H': (ALARM:8) Homing fail. Pull off travel failed to clear limit switch. Try increasing pull-off setting or check wiring. Streaming has been paused.
[Error] Error while processing response <An error was detected while sending '$H': (ALARM:8) Homing fail. Pull off travel failed to clear limit switch. Try increasing pull-off setting or check wiring. Streaming has been paused.>
[Error] Error while processing response <ok>: An unexpected command was completed by the controller.
GrblHAL 1.1f ['$' for help]
>>> $$
$0 = 10.0    (Step pulse time, microseconds)
$1 = 25    (Step idle delay, milliseconds)
$2 = 7    (Step pulse invert, mask)
$3 = 0    (Step direction invert, mask)
$4 = 7    (Invert step enable pin, boolean)
$5 = 0    (Invert limit pins, boolean)
$6 = 0    (Invert probe pin, boolean)
$10 = 2    (Status report options, mask)
$11 = 0.010    (Junction deviation, millimeters)
$12 = 0.002    (Arc tolerance, millimeters)
$13 = 0    (Report in inches, boolean)
$14 = 64   
$15 = 0   
$16 = 0   
$17 = 0   
$18 = 0   
$19 = 0   
$20 = 0    (Soft limits enable, boolean)
$21 = 1    (Hard limits enable, boolean)
$22 = 1    (Homing cycle enable, boolean)
$23 = 0    (Homing direction invert, mask)
$24 = 25.000    (Homing locate feed rate, mm/min)
$25 = 500.000    (Homing search seek rate, mm/min)
$26 = 25    (Homing switch debounce delay, milliseconds)
$27 = 1.000    (Homing switch pull-off distance, millimeters)
$28 = 0.100   
$29 = 0.0   
$30 = 1000.    (Maximum spindle speed, RPM)
$31 = 0.    (Minimum spindle speed, RPM)
$32 = 0    (Laser-mode enable, boolean)
$33 = 5000.000   
$34 = 0.000   
$35 = 0.000   
$36 = 100.000   
$37 = 0   
$39 = 1   
$40 = 0   
$41 = 0   
$42 = 2   
$43 = 1   
$44 = 4   
$45 = 3   
$46 = 0   
$56 = 5.000   
$57 = 100.000   
$58 = -5.000   
$59 = 500.000   
$60 = 0   
$61 = 0   
$62 = 0   
$63 = 2   
$64 = 0   
$65 = 0   
$70 = 3   
$100 = 158.302    (X-axis travel resolution, step/mm)
$101 = 159.400    (Y-axis travel resolution, step/mm)
$102 = 159.453    (Z-axis travel resolution, step/mm)
$110 = 500.000    (X-axis maximum rate, mm/min)
$111 = 500.000    (Y-axis maximum rate, mm/min)
$112 = 800.000    (Z-axis maximum rate, mm/min)
$120 = 10.000    (X-axis acceleration, mm/sec^2)
$121 = 10.000    (Y-axis acceleration, mm/sec^2)
$122 = 10.000    (Z-axis acceleration, mm/sec^2)
$130 = 200.000    (X-axis maximum travel, millimeters)
$131 = 200.000    (Y-axis maximum travel, millimeters)
$132 = 200.000    (Z-axis maximum travel, millimeters)
$300 = GRBL   
$301 = 0   
$302 = 10.1.1.2   
$303 = 10.1.1.1   
$304 = 255.255.255.0   
$305 = 23   
$307 = 80   
$341 = 0   
$342 = 30.0   
$343 = 25.0   
$344 = 200.0   
ok
>>> $G
[verbose] GrblFeedbackMessage{message='[GC:G0 G54 G17 G21 G90 G94 G49 G98 G50 M5 M9 T0 F0 S0.]', distanceMode='G90', units='G21'}
[GC:G0 G54 G17 G21 G90 G94 G49 G98 G50 M5 M9 T0 F0 S0.]
ok
[verbose] <Alarm|WPos:0.000,0.000,0.000|Bf:35,1023|H:0>
[verbose] <Alarm|WPos:0.000,0.000,0.000|Bf:35,1023>

**Reply by terjeio** [reply]
> I did some editing to reduce repeated data so hopefully this caught what you asked for

It did not, can you try with setting `$10=511` and `$5=7`?
If this does not help post the output from `$I`.

Be aware that grblHAL (and grbl) does not respond to real-time report requests during homing and due to that the sender cannot show any movement in the DRO display.

**Reply by jrdarrah** [reply]
The $5=7 did the trick. I dug out the spec sheet for the opto-isolators that Phil used on his grblHAL breakout board. My proximity sensors are normally open so they would read 5V because of the pull-up resistor on the board's interface.  The spec shows 5V in gives a low on the isolators output to the Teensy pin.  Inverting the limit pins via $5=7 fixed that issue. I was not aware that  the DRO wouldn't change during the homing cycle.  I went back and connected all stepper motors.  The home cycle did move each axis to the proper position.  The pull off looks about right. Thanks.

**Comment by WZdunski** [comment]
HI, 
After uploading, there is an error: 9, 18. I changed $ 14 to 7 and the alarm disappears.
But the problem I am struggling with is connecting the motors. After connecting the Protoneer v3 CNC board, there is always power on the motors control pins. I have implemented every map and this problem is always there.
What should I change to solve this problem?

My staff:
Protoneer CNC Shield V3 (red)
Arduino Due Atmel SAM3X8E ARM Cortex-M3 CPU

Thanks for any advice.

**Reply by phil-barrett** [reply]
What does "always power" mean?  If the stepper driver is not disabled, there should be power on the motor output pins.

**Reply by WZdunski** [reply]
On all pins I have power on one time. 
So the motor is not moveable.

![IMG_20220102_165037](https://user-images.githubusercontent.com/96832902/147883800-67c52bbd-d1d9-4b23-9d1d-32a757c1e3a1.jpg)

**Reply by WZdunski** [reply]
$0=10.0
$1=25
$2=3
$3=0
$4=1
$5=6
$6=0
$7=1
$10=0
$11=0.010
$12=0.002
$13=0
$14=7
$15=0
$16=0
$17=1
$18=0
$19=0
$20=0
$21=0
$22=0
$23=0
$24=25.0
$25=500.0
$26=250
$27=1.000
$28=0.100
$29=0.0
$30=1000.000
$31=0.000
$32=1
$33=5000.0
$34=0.0
$35=0.0
$36=100.0
$37=0
$39=0
$40=0
$43=1
$44=4
$45=3
$46=0
$62=0
$63=3
$64=0
$65=0
$80=1.000
$81=0.010
$82=0.000
$84=0.000
$85=10.000
$90=0.000
$91=0.000
$92=0.000
$95=0.000
$100=250.000
$101=250.000
$102=250.000
$110=500.000
$111=500.000
$112=500.000
$120=10.000
$121=10.000
$122=10.000
$130=200.000
$131=200.000
$132=200.000
$341=0
$342=30.0
$343=25.0
$344=200.0
$345=100.0
$347=5.0
$348=2.500
$349=25.000
ok
_________________________

[GC:G0 G54 G17 G21 G90 G94 G49 G98 G50 M5 M9 T0 F0 S0.]
_________________________

[VER:1.1f.20210513:]
[OPT:VN0SLW,35,1024,3,0]
[NEWOPT:ENUMS,RT-,NOPROBE,TC]
[FIRMWARE:grblHAL]
[NVS STORAGE:*FLASH]
[DRIVER:SAM3X8E]
[DRIVER VERSION:210503]
[BOARD:Ramps 1.6]
_________________________

**Reply by phil-barrett** [reply]
It is still not clear to me what you mean by "power on one time."

Do you have the motors wired correctly? (A+ and A- to a single coil, same for B+ and B-).  To find the leads for a single coil, short a pair of leads together. If the motor gets harder to turn by hand, those leads are for a single coil. If not, they are for different coils.

I would break it down into each section. Is grblHAL generating the right signals on step, dir and en? (pull the shield off for direct access to the pins). Are step, dir and en for the axis correctly connected to the driver and getting the correct signals? (pull the drivers out for access). Are the motor pins getting sequenced correctly?  You can use an LED+resistor to visually see what is going on at each stage.

**Reply by WZdunski** [reply]
When I connect the motors, the motors is short-circuited because current flows on A and B at the same time. As you can see, the board is not connected to anything except the power supply and the USB socket. After connecting the USB and enabling communication with the device, the pins show the current, i.e. A (green and blue) B (yellow and orange), I do not send any work. As soon as I unplug the USB cable, the power disappears from the pins A,B.

**Reply by phil-barrett** [reply]
The motor coils should be getting current unless the enable signal is not asserted. (It should be called "disable" signal.)  In a lot of the Grbl shields, enable isn't even brought to the driver so the drivers should be outputting current all the time.  That's how they hold position.  Removing the USB cable powers down the board and probably the shield which would cause the drivers to drop current. Have you tried jogging the axis for the connected motor?

**Reply by WZdunski** [reply]
No axis moves when I by the program asks for a shift. all motors are tightened.
How I can to OFF signal?

**Reply by terjeio** [reply]
Set $4=7 to invert stepper enable pins, then measure voltage on the EN pin (black circle). It should be 0V when moving motors, 3.3V when not. When 3.3V you should be able to turn the motor easily by hand.

![image](https://user-images.githubusercontent.com/20260062/147888675-7ae96c50-aa7f-40e0-b62e-4767edb361a4.png)

The Z-axis driver has to be moved to the yellow socket.

I have cut the 5V pin on my shield and shorted it to the 3.3V pin as the Due is a 3.3V processor. I guess you may damage the Due if not doing that.

![image](https://user-images.githubusercontent.com/20260062/147888950-6c6eb560-32e5-4685-8227-d56a18f780c8.png)

**Reply by WZdunski** [reply]
Whan I try to change a $4 to 7 i get this:

[ ] error: 1 - G-code words consist of a letter and a value. Letter was not found. [ $4=7 ]

[20:54:34] [ ERROR ] error: 1 - G-code words consist of a letter and a value. Letter was not found. [ $4=7 ]

[20:54:34] [ $4=7 ] error:1

**Reply by WZdunski** [reply]
Im dont use  Z axes, The driver in red options is for copy Y Axes.

**Reply by WZdunski** [reply]
Does it matter which map I choose? https://github.com/grblHAL/SAM3X8E
 So should cut the pin you show in the picture?

**Reply by phil-barrett** [reply]
Yes, it is important to use the correct map for your breakout board.  Do you have pin assignments for the Protoneer v3 CNC board? Other breakout boards are different and need different pin map.

**Reply by terjeio** [reply]
There is a map for the [board in the new repo](https://github.com/grblHAL/SAM3X8E/blob/af0ac00b1257c89280956e945f23c5253a72c60c/grblHAL_Due/src/my_machine.h#L22-L31), I cannot remember if there is one in this and I am lazy to check as I am about to archive it.

> So should cut the pin you show in the picture?

Yes, and solder a bridge between 3.3V and 5V on the shield as shown.

Phil, can you update the link to the new repo on your Tindie page and also in grbl.org if still pointing to this?

**Reply by WZdunski** [reply]
Even when I choose the Protoneer map, I have RAMPS1.6 in the description. So I'm not sure it will load.
 
[VER:1.1f.20210513:]
[OPT:VNSL,35,1024,3,0]
Target buffer size found
[NEWOPT:ENUMS,RT+,NOPROBE,TC]
[FIRMWARE:grblHAL]
[NVS STORAGE:*FLASH]
[DRIVER:SAM3X8E]
[DRIVER VERSION:210503]
[BOARD:Ramps 1.6]


![image](https://user-images.githubusercontent.com/96832902/147891556-051499ee-606f-4404-9775-c116339c6d10.png)

**Reply by WZdunski** [reply]
Please, tell me how I should set $ for laser with two axis XY (Y double) 
I uploaded https://github.com/grblHAL/SAM3X8E with map Protoneer

My current settings..
$14 i set 73 but it showing 1

GrblHAL 1.1f ['$' or '$HELP' for help]
<Idle|WPos:0.000,0.000,0.000|Bf:35,1023|WCS:G54|A:|Sc:|H:0,0|T:0|TLR:0|FW:grblHAL>
[VER:1.1f.20210513:]
[OPT:VNH0SLW,35,1024,3,0]
[NEWOPT:ENUMS,RT+,HOME,NOPROBE,TC]
[FIRMWARE:grblHAL]
[NVS STORAGE:*FLASH]
[DRIVER:SAM3X8E]
[DRIVER VERSION:210503]
[BOARD:Ramps 1.6]
ok
$0=10.0
$1=25
$2=7
$3=7
$4=7
$5=0
$6=0
$7=1
$10=2
$11=0.010
$12=0.002
$13=0
$14=1
$15=0
$16=1
$17=0
$18=0
$19=0
$20=0
$21=0
$22=131
$23=7
$24=25.0
$25=500.0
$26=250
$27=1.000
$28=0.100
$29=0.5
$30=1000.000
$31=0.000
$32=1
$33=5000.0
$34=0.0
$35=0.0
$36=100.0
$37=7
$39=1
$40=1
$43=1
$44=4
$45=3
$46=0
$62=0
$63=3
$64=0
$65=0
$80=1.000
$81=0.010
$82=0.000
$84=0.000
$85=10.000
$90=0.000
$91=0.000
$92=0.000
$95=0.000
$100=250.000
$101=250.000
$102=250.000
$110=500.000
$111=500.000
$112=500.000
$120=10.000
$121=10.000
$122=10.000
$130=200.000
$131=200.000
$132=200.000
$341=0
$342=30.0
$343=25.0
$344=200.0
$345=100.0
$347=5.0
$348=2.500
$349=25.000
ok
[GC:G0 G54 G17 G21 G90 G94 G49 G98 G50 M5 M9 T0 F0 S0.]
ok
[G54:0.000,0.000,0.000]
[G55:0.000,0.000,0.000]
[G56:0.000,0.000,0.000]
[G57:0.000,0.000,0.000]
[G58:0.000,0.000,0.000]
[G59:0.000,0.000,0.000]
[G59.1:0.000,0.000,0.000]
[G59.2:0.000,0.000,0.000]
[G59.3:0.000,0.000,0.000]
[G28:0.000,0.000,0.000]
[G30:0.000,0.000,0.000]
[G92:0.000,0.000,0.000]
[HOME:0.000,0.000,0.000:0]
[TLO:0.000,0.000,0.000]
[PRB:0.000,0.000,0.000:0]

**Reply by phil-barrett** [reply]
$32=1 is laser mode.

Have you tried [ioSender](https://github.com/terjeio/ioSender/releases/tag/2.0.35)?  Only runs on a PC but is a very good GCode Sender.

**Reply by WZdunski** [reply]
Laser mode i have ON. Any else change on $? 
Today i installed the iosender. 
If i cut the pin 5V and i will do a bridge should working the shield?

**Comment by phil-barrett** [comment]
> Phil, can you update the link to the new repo on your Tindie page and also in grbl.org if still pointing to this?

I fixed several links to the old repo. Will go looking for others.

**Reply by WZdunski** [reply]
Unfortunately, the option with cutting the 5V pin and making the 3.3V bridge does not work. I still have POWER on the motor control pins A(12v) B(12v) on one time. Changing $ 4 doesn't help. I have no current on EN pins but it hasn't changed anything.

**Reply by terjeio** [reply]
To enable the Protoneer map you have to change board selection like this:

```
// If none is enabled pin mappings from generic_map.h will be used
// NOTE: Only one board may be enabled!
//#define BOARD_TINYG2_DUE
//#define BOARD_RAMPS_16
//#define BOARD_MEGA256
#define BOARD_PROTONEER
//#define BOARD_CMCGRATH
//#define BOARD_RADDS_16   // NOTE: untested!!
//#define BOARD_MY_MACHINE // Add my_machine_map.h before enabling this!
```

If not the correct map is selected it is highly unlikely that anything will work.

BTW currently the Protoneer board map does not support a ganged axis - but I believe it could be added.

**Reply by WZdunski** [reply]
Yes, I changed and I selected the board Protoneer. But I said before https://github.com/terjeio/grblHAL/discussions/291#discussioncomment-1897435 
still in description is [BOARD:Ramps 1.6]

**Reply by WZdunski** [reply]
What that mean ? "not support a ganged axis  ?

**Reply by terjeio** [reply]
> Yes, I changed and I selected the board Protoneer. But I said before #291 (reply in thread)
still in description is [BOARD:Ramps 1.6]

This means the correct board is _not_ compiled in - check again. If `#define BOARD_RAMPS_16` does not have `//` in front of it it will be selected over the Protoneer map which is later in the list.

> What that mean ? "not support a ganged axis ?

 You wrote earlier "The driver in red options is for copy Y Axes." - if you copy Y-axis by hardware links you can ignore my comment.

**Reply by WZdunski** [reply]
Im sorry but Im comfused! https://github.com/terjeio/grblHAL/discussions/291#discussioncomment-1898590 You said if "//" is off in front PROTONEER it should load to device.. but now you saying it should off "//" in BOARD_RAMPS_16.. 
> This means the correct board is _not_ compiled in - check again. If `#define BOARD_RAMPS_16` does not have `//` in front of it it will be selected over the Protoneer map which is later in the list.
> 
> > Yes, I changed and I selected the board Protoneer. But I said before #291 (reply in thread)
> > still in description is [BOARD:Ramps 1.6]
>

**Reply by phil-barrett** [reply]
In the reply you linked to, you will see exactly how it should be. The "//" tell the compiler to ignore everything to the right. So in the code that Terje showed is telling the compiler to only use #define PROTONEER

If it says [BOARD: RAMPS 1.6] in the description, then not actually using the modified file?

**Reply by WZdunski** [reply]
So like you see https://github.com/terjeio/grblHAL/discussions/291#discussioncomment-1897435 I took off "//" in Protoneer but still it showing [BOARD:Ramps 1.6] after connected. 

And I see in driver.h is changed but in softwares in I have RAMPS1.6 in the description.

So I don't know what I doing wrong... 

 [VER:1.1f.20210513:]
[OPT:VNSL,35,1024,3,0]
Target buffer size found
[NEWOPT:ENUMS,RT+,NOPROBE,TC]
[FIRMWARE:grblHAL]
[NVS STORAGE:*FLASH]
[DRIVER:SAM3X8E]
[DRIVER VERSION:210503]
[BOARD:Ramps 1.6]

[Screenshot 2022-01-03 at 20 32 41](https://user-images.githubusercontent.com/96832902/147977370-1c0c1fdf-f2c1-4c42-a7fb-8a5453451906.png)

**Reply by phil-barrett** [reply]
I'm not familiar with the Due driver. How is mymachine.h used?  In other drivers, it selects the board. Is that where the Ramps 1.6 selection is happening?

**Reply by WZdunski** [reply]
Even when I choose other maps and uploaded, it always shows up the Ramps 1.6.

**Reply by phil-barrett** [reply]
open mymachine.h 
It may have something similar to what Terje posted.

**Reply by terjeio** [reply]
> So I don't know what I doing wrong...

```
[VER:1.1f.20210513:]
[DRIVER VERSION:210503]
```

You are not compiling the version from the [new repo](https://github.com/grblHAL/SAM3X8E), this is what I get. 

```
[VER:1.1f.20211221:]
[DRIVER VERSION:211213]
[BOARD:Protoneer v3]
```

I am sorry for not spotting that before - too much going on...

**Reply by WZdunski** [reply]
Please look on my screenshots.
![](https://user-images.githubusercontent.com/96832902/147977370-1c0c1fdf-f2c1-4c42-a7fb-8a5453451906.png)

**Reply by terjeio** [reply]
Are you editing a different version of the code than the compiler uses? The version strings tells me so.

**Reply by WZdunski** [reply]
> Are you editing a different version of the code than the compiler uses? The version strings tells me so.

Small SUCCESS ! 
After clean the cesh, I got correct description. And motors pins don't have a power like before.. 

[VER:1.1f.20211226:]
[OPT:VNMSL,35,1024,3,0]
Target buffer size found
[NEWOPT:ENUMS,RT+,TC,SED]
[FIRMWARE:grblHAL]
[NVS STORAGE:*FLASH]
[DRIVER:SAM3X8E]
[DRIVER VERSION:211213]
[BOARD:Protoneer v3]


**For now very thanks for your persistence!**
